sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_deployment.py

```diff
@@ -194,6 +195,11 @@ def _yaml_output_action_to_rest_output_action(cls, yaml_output_action: str) -> s
         return output_switcher.get(yaml_output_action, yaml_output_action)
 
     def _to_rest_object(self, location: str) -> BatchDeploymentData:  # pylint: disable=arguments-differ
+        warnings.warn(
```

Comment:
Shouldn't we have this warning in the constructor so that customers will be warned the moment they attempt to use this object?